compiler/noirc_frontend/src/parser/parser.rs

```diff
@@ -1082,6 +1084,12 @@ fn named_type(type_parser: impl NoirParser<UnresolvedType>) -> impl NoirParser<U
         .map_with_span(|(path, args), span| UnresolvedTypeData::Named(path, args).with_span(span))
 }
 
+fn named_trait(type_parser: impl NoirParser<UnresolvedType>) -> impl NoirParser<UnresolvedType> {
+    keyword(Keyword::Impl).then(path()).then(generic_type_args(type_parser)).map_with_span(
+        |((_token, path), args), span| UnresolvedTypeData::TraitAsType(path, args).with_span(span),
+    )
```

Comment:
The suggested code does not compile because `path` is lost when `ignore_then` is used. In the next PR I have prepared, where I implemented `fn() -> impl TraitA + TraitB` this whole function is rewritten. So let's leave it as it is in this PR.

Reply:
Fixed